prov/verbs/src/windows/infiniband/verbs.h

```diff
@@ -0,0 +1,3565 @@
+ * effect of an application calling fork() is undefined.
+ */
+int ibv_fork_init(void);
+
+/**
+ * ibv_node_type_str - Return string describing node_type enum value
+ */
+const char *ibv_node_type_str(enum ibv_node_type node_type);
+
+/**
+ * ibv_port_state_str - Return string describing port_state enum value
+ */
+const char *ibv_port_state_str(enum ibv_port_state port_state);
+
+/**
+ * ibv_event_type_str - Return string describing event_type enum value
+ */
+const char *ibv_event_type_str(enum ibv_event_type event);
+
+#define ETHERNET_LL_SIZE 6
+int ibv_resolve_eth_l2_from_gid(struct ibv_context *context,
+				struct ibv_ah_attr *attr,
+				uint8_t eth_mac[ETHERNET_LL_SIZE],
+				uint16_t *vid);
+
+/*static inline*/ int ibv_is_qpt_supported(uint32_t caps, enum ibv_qp_type qpt);
+/*
+{
+	return !!(caps & (1 << qpt));
+}
+*/
+
+/*static inline*/ struct ibv_counters *ibv_create_counters(struct ibv_context *context,
+						       struct ibv_counters_init_attr *init_attr);
+/*
+{
+	struct verbs_context *vctx;
+
+	vctx = verbs_get_ctx_op(context, create_counters);
+	if (!vctx) {
+		errno = EOPNOTSUPP;
+		return NULL;
+	}
+
+	return vctx->create_counters(context, init_attr);
+}
+*/
+
+/*static inline*/ int ibv_destroy_counters(struct ibv_counters *counters);
+/*
+{
+	struct verbs_context *vctx;
+
+	vctx = verbs_get_ctx_op(counters->context, destroy_counters);
+	if (!vctx)
+		return EOPNOTSUPP;
+
+	return vctx->destroy_counters(counters);
+}
+*/
+
+/*static inline*/ int ibv_attach_counters_point_flow(struct ibv_counters *counters,
+						 struct ibv_counter_attach_attr *attr,
+						 struct ibv_flow *flow);
+/*
+{
+	struct verbs_context *vctx;
+
+	vctx = verbs_get_ctx_op(counters->context, attach_counters_point_flow);
+	if (!vctx)
+		return EOPNOTSUPP;
+
+	return vctx->attach_counters_point_flow(counters, attr, flow);
+}
+*/
+
+/*static inline*/ int ibv_read_counters(struct ibv_counters *counters,
+				    uint64_t *counters_value,
+				    uint32_t ncounters,
+				    uint32_t flags);
+/*
+{
+	struct verbs_context *vctx;
+
+	vctx = verbs_get_ctx_op(counters->context, read_counters);
+	if (!vctx)
+		return EOPNOTSUPP;
+
+	return vctx->read_counters(counters, counters_value, ncounters, flags);
+}
+*/
+
+#define IB_ROCE_UDP_ENCAP_VALID_PORT_MIN (0xC000)
+#define IB_ROCE_UDP_ENCAP_VALID_PORT_MAX (0xFFFF)
+#define IB_GRH_FLOWLABEL_MASK (0x000FFFFF)
+
+/*static inline*/ uint16_t ibv_flow_label_to_udp_sport(uint32_t fl);
+/*
+{
+	uint32_t fl_low = fl & 0x03FFF, fl_high = fl & 0xFC000;
+
+	fl_low ^= fl_high >> 14;
+	return (uint16_t)(fl_low | IB_ROCE_UDP_ENCAP_VALID_PORT_MIN);
+}
+*/
+
+/**
+ * ibv_set_ece - Set ECE options
+ */
+int ibv_set_ece(struct ibv_qp *qp, struct ibv_ece *ece);
+
+/**
+ * ibv_query_ece - Get accepted ECE options
+ */
+int ibv_query_ece(struct ibv_qp *qp, struct ibv_ece *ece);
+#ifdef __cplusplus
+}
+#endif
+
+#  undef __attribute_const
+
+
+#endif /* INFINIBAND_VERBS_H */
```

Comment:
See my comment above about whether you'd like me to invest some time in pruning the imported header files.

As it stands, I only provided implementations for methods that the verbs provider actually uses and even a handful of those simply return ENOSYS at the moment.

If a declared function that isn't implemented on Windows were to be added to the verbs code, it would result in a link error due to an undefined symbol. Not quite a compilation failure, but close.